### ISSUE TITLE
Maximum call stack size exceeded (from candidatePermutations in generateRules.js)

### ISSUE THREAD

**Issue by frading** [rails/tailwindcss-rails#154]
Hello,

I am integrating tailwindcss-rails into a rails 6 app, and am getting the following error:

```
> bin/rails tailwindcss:install
W, [2022-03-09T10:41:22.420190 #2029393]  WARN -- Skylight: [SKYLIGHT] [5.1.1] Running Skylight in development mode. No data will be reported until you deploy your app.
(To disable this message for all local apps, run `skylight disable_dev_warning`.)
Default application.html.erb is missing!
        Add <%= stylesheet_link_tag "tailwind", "inter-font", "data-turbo-track": "reload" %> within the <head> tag in your custom layout.
Build into app/assets/builds
       exist  app/assets/builds
   identical  app/assets/builds/.keep
File unchanged! The supplied flag value not found!  app/assets/config/manifest.js
File unchanged! The supplied flag value not found!  .gitignore
File unchanged! The supplied flag value not found!  Procfile.dev
Add bin/dev to start foreman
   identical  bin/dev
Compile initial Tailwind build
         run  rails tailwindcss:build from "."
+ /home/gui/.rbenv/versions/3.0.3/lib/ruby/gems/3.0.0/gems/tailwindcss-rails-2.0.7-x86_64-linux/exe/x86_64-linux/tailwindcss -i /home/gui/work/web/polygonjs/dev/polygonjs-editor/app/assets/stylesheets/application.tailwind.css -o /home/gui/work/web/polygonjs/dev/polygonjs-editor/app/assets/builds/tailwind.css -c /home/gui/work/web/polygonjs/dev/polygonjs-editor/config/tailwind.config.js --minify
/snapshot/tailwindcss/lib/lib/generateRules.js:36
function* candidatePermutations(candidate, lastIndex = Infinity) {
                               ^

RangeError: Maximum call stack size exceeded
    at candidatePermutations (/snapshot/tailwindcss/lib/lib/generateRules.js:36:32)
    at candidatePermutations.next (<anonymous>)
    at candidatePermutations (/snapshot/tailwindcss/lib/lib/generateRules.js:61:12)
    at candidatePermutations.next (<anonymous>)
    at candidatePermutations (/snapshot/tailwindcss/lib/lib/generateRules.js:61:12)
    at candidatePermutations.next (<anonymous>)
    at candidatePermutations (/snapshot/tailwindcss/lib/lib/generateRules.js:61:12)
    at candidatePermutations.next (<anonymous>)
    at candidatePermutations (/snapshot/tailwindcss/lib/lib/generateRules.js:61:12)
    at candidatePermutations.next (<anonymous>)
```

I initially thought it would be because the tailwindcss executable did not include the fix from https://github.com/tailwindlabs/tailwindcss/pull/7331, which was released in version 3.0.19, but then I see here https://github.com/rails/tailwindcss-rails/pull/143 that it may indeed be in the executable.

Any idea of what I could do to debug this?

**Comment by frading** [comment]
Still investigating this, I've just locally replaced the executable shipped with the gem, with the one from tailwind v3.0.23 (https://github.com/tailwindlabs/tailwindcss/releases/tag/v3.0.23) and that seems to have fixed the issue.

**Comment by dhh** [comment]
Included 3.0.23 in 2.0.8 👍
